src/components/register/RegisterUser.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import 'react-toastify/dist/ReactToastify.css';
+import { fetchUsers } from '../../actions/registerUserActions';
+
+export class RegisterUser extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      username: '',
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.registerUserSuccess === true) {
+      this.props.history.push('/');
+    }
+  }
+
+  handleChange = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleSubmit = event => {
+    if (event) event.preventDefault();
+    const payload = {
+      user: {
+        email: this.state.email,
+        password: this.state.password,
+        username: this.state.username
+      },
+    };
+
+    this.props.fetchUsers(payload);
+  }
+
+  render() {
+    const { email, password, username } = this.state;
+    return (
+      <div className="registerForm">
+        <h1>Register</h1>
+        <form id="form" className="registerUser" onSubmit={this.handleSubmit}>
+          <div className="RegisterForm__input">
+            <label htmlFor="email">Email</label>
+            <br />
+            <input type="text" name="email" id="email" value={email} onChange={this.handleChange} />
+          </div>
+          <div className="RegisterForm__input">
+            <label htmlFor="password">Password</label>
+            <br />
+            <input type="password" name="password" id="password" value={password} onChange={this.handleChange} />
+          </div>
+          <div className="RegisterForm__input">
+            <label htmlFor="username">Username</label>
+            <br />
+            <input type="text" name="username" id="username" value={username} onChange={this.handleChange} />
+          </div>
+          <button className="btn" type="submit" value="Submit">Submit</button>
+        </form>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  registerUserSuccess: state.user.registerUserSuccess,
+  registerUserError: state.user.registerUserError,
```

Comment:
Do you still need this?

Reply:
I left that there just in the event that errors are shown within the form